webui/src/pages/repositories/repository/objectViewer.tsx

```diff
@@ -109,98 +61,56 @@ export const getContentType = (headers: Headers): string | null => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const sizeBytes = parseInt((response as any)?.headers.get('Content-Length'))
+        content = <FileContents
+            repoId={repoId}
+            refId={refId}
+            path={path}
+            fileExtension={fileExtension}
+            contentType={contentType}
+            sizeBytes={sizeBytes}
+            error={error}
+            loading={loading}
+        />
     }
 
-    // We'll need to convert the API service to get rid of this any
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const responseText = (response as any)?.responseText;
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const responseBlob = (response as any)?.responseBlob;
     return (
         <RefContextProvider>
             <RepositoryPageLayout activePage={'objects'}>
-                {loading && <Loading/>}
-                <FileContents 
-                    repoId={repoId} 
-                    refId={refId}
-                    path={decodedObjectName}
-                    fileExtension={fileExtension}
-                    contentType={contentType}
-                    rawContent={responseText}
-                    blobContent={responseBlob}
-                    error={error}
-                    loading={loading}
-                />
+                {content}
             </RepositoryPageLayout>
         </RefContextProvider>
     );
 };
 
-export const FileContents: FC<FileContentsProps> = ({repoId, refId, path, loading, error, rawContent, blobContent, contentType = null, fileExtension='', showFullNavigator = true}) => {
+export const FileContents: FC<FileContentsProps> = ({repoId, refId, path, loading, error, contentType = null, fileExtension='', sizeBytes, showFullNavigator = true}) => {
 
     const objectUrl = linkToPath(repoId, refId, path);
 
     if (loading || error) {
         return <></>;
     }
 
-    const size = (rawContent) ? rawContent.length : blobContent.size;
-
-    let content;
-    const renderer = resolveRenderer(size, contentType, fileExtension);
-    if (!renderer) {
-        // let's try and guess?
-        if (size <= MAX_DISPLAYABLE_SIZE) {
-            const language = guessLanguage(fileExtension, contentType);
-            if (language) {
-                content = <GenericRenderer content={rawContent} language={language}/>;
-            } else {
-                // File type does not have a renderer
-                // We cannot display it inline
-                content = <FileTypeNotSupported />
-            }
-        } else {
-            // File type does not have a renderer
-            // We cannot display it inline
-            content = <FileTypeNotSupported />
-        }
-    } else {
-        content = React.createElement(renderer, { content: rawContent, contentBlob: blobContent, }, []);
-    }
-
     const repo = {
         id: repoId,
     };
```

Comment:
Since you changed the URL, I suggest adding the filename value to the `download` prop of the download button, so you get the correct filename in the download

Reply:
True. Done